### PR TITLE
test: Add new e2e test suite for v2prov

### LINKED ISSUE #106: [e2e] Add test to ensure v2prov doesn't break
### What would you like to be added (User Story)?

As a user of Rancher Turtles
I want the option of using different provisioning mechanisms in Rancher
So that i can pick and choose depending on my needs

### Detailed Description

We'll need to add a unit test where:
- Rancher is srated with Rancher Turtles (i.e. embedded capi feature disabled and operators deployed)
- Create a cluster using v2prov
- Ensure cluster is created without an issue

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind feature

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

This adds a new e2e test suite that will ensure that v2prov doesn't break with Rancher Turtles installed.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #106 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [x] adds or updates e2e tests

### CODE REVIEW

**Review by salasberryfin — APPROVED** [review]

**Review by Danil-Grigorev — COMMENTED** [review]
Does this e2e require additional variables in the nightly job?

File: test/e2e/data/rancher/rancher-service-patch.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ apiVersion: v1
 kind: Service
 metadata:
   annotations:
-    k8s.ngrok.com/app-protocols: '{"https-internal":"HTTPS","http":"HTTP"}'
```
Comment (Danil-Grigorev) [inline test/e2e/data/rancher/rancher-service-patch.yaml:5]: I’m not convinced this change is required. As I mentioned before, this annotations are just mapping to the port names in the service. Additionally in my setup now the ui via RANCHER_HOSTNAME is not accessible anymore.

Comment (richardcase) [reply test/e2e/data/rancher/rancher-service-patch.yaml:5]: It works for me. But let me change this back and try again.....will report back shortly 😄

Comment (Danil-Grigorev) [reply test/e2e/data/rancher/rancher-service-patch.yaml:5]: Using current setup neither AWS nor Azure cluster had no issue with connecting agent to the rancher cluster. Here is a screenshot of fully imported AKS cluster https://github.com/rancher-sandbox/rancher-turtles/pull/133#issuecomment-1743339541

Comment (richardcase) [reply test/e2e/data/rancher/rancher-service-patch.yaml]: That worked with reverting it, thanks @Danil-Grigorev

File: test/e2e/const.go [hunk 2]
```diff
@@ -83,7 +89,12 @@ const (
 	GiteaUserNameVar     = "GITEA_USER_NAME"
 	GiteaUserPasswordVar = "GITEA_USER_PWD"
 
+	RKE2VersionVar = "RKE2_VERSION"
+
 	CapaEncodedCredentialsVar = "CAPA_ENCODED_CREDS"
+	AzureSubIDVar             = "AZURE_SUBSCRIPTION_ID"
+	AzureClientIDVar          = "AZURE_CLIENT_ID"
+	AzureClientSecretVar      = "AZURE_CLIENT_SECRET"
```
Comment (Danil-Grigorev) [inline test/e2e/const.go:97]: This 3 variables should be added to env in `e2e-long.yaml` so the post submit and nightly won’t fail after this gets merged.

**Review by Danil-Grigorev — APPROVED** [review]

File: test/e2e/const.go (hunk 2 above)
Comment (richardcase) [reply test/e2e/const.go:97]: These have been added to the GHA secrets.

Comment (richardcase) [reply test/e2e/const.go:97]: Yes you are right, adding now.

**Review by furkatgofurov7 — APPROVED** [review]
LGTM

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**